lib/email_alert_signup_api.rb

```diff
@@ -1,20 +1,34 @@
+require 'uri'
+
 class EmailAlertSignupAPI
-  def initialize(applied_filters:, default_filters:, facets:, subscriber_list_title:, finder_format:)
+  def initialize(applied_filters:, default_filters:, facets:, subscriber_list_title:, finder_format:, default_frequency: nil)
     @applied_filters = applied_filters.deep_symbolize_keys
     @default_filters = default_filters.deep_symbolize_keys
     @facets = facets
     @subscriber_list_title = subscriber_list_title
     @finder_format = finder_format
+    @default_frequency = default_frequency
   end
 
   def signup_url
-    subscriber_list['subscription_url']
+    if @default_frequency
+      add_url_param(subscriber_list['subscription_url'], default_frequency: @default_frequency)
+    else
+      subscriber_list['subscription_url']
+    end
   end
 
 private
 
   attr_reader :applied_filters, :default_filters, :facets, :subscriber_list_title, :finder_format
 
+  def add_url_param(url, param)
+    # this method safely adds a URL parameter using the correct one of '?' or '&'
+    utm_uri = Addressable::URI.parse(url)
```

Comment:
`utm_uri` is presumably copied from elsewhere?  Could this be renamed to something else, since the `utm` part relates to Google Analytics's campaign tracking functionality.

Reply:
Whoops